core/src/queue.rs

```diff
@@ -115,6 +117,37 @@ impl Queue {
             .collect()
     }
 
+    /// Returns latest asset registering transaction hash.
+    pub fn latest_asset_reg_tx_hash(
+        &self,
+        wsv: &WorldStateView,
+    ) -> Option<HashOf<VersionedSignedTransaction>> {
+        self.txs
+            .iter()
+            .filter(|e| -> bool {
+                self.is_pending(e.value(), wsv)
+                    && match e.value().payload().instructions {
+                        Executable::Instructions(ref instructions) => {
+                            instructions.iter().any(|i| match i {
+                                Instruction::Register(reg_box) => {
+                                    let context = Context::new();
+                                    reg_box.object.evaluate(wsv, &context).map_or(
+                                        false,
+                                        |reg_box| {
+                                            matches!(reg_box, RegistrableBox::AssetDefinition(_))
+                                        },
+                                    )
+                                }
+                                _ => false,
+                            })
+                        }
+                        Executable::Wasm(_) => false,
```

Comment:
you know wasm can also register an asset?

Reply:
this is going to get way more complicated

Reply:
Created #3357 and left a TODO there in case it might be big.

Reply:
for me this is a big no no. What is the point of introducing a feature that you know doesn't work? You could say that you made half the work done, but properly handling this might require a code that is very different than the current code in which case the current code is and would be useless. Just in case that someone would think I'm nitpicking I would like to get opinions from other colleagues

Reply:
if you remove the query than I'd be ok with merging the code as is because then it cannot be misused